vendor/radixtree/ffi_radix_tree.cpp

```diff
@@ -53,6 +52,33 @@ const char* longest_prefix(radix_tree<std::string, std::vector<char>>* map_point
   return NULL;
 }
 
+const char* longest_prefix_and_value(radix_tree<std::string, std::vector<char>>* map_pointer, const char* key, int* read_size, int* prefix_size) {
+  std::string string_key(key);
+  auto iter = map_pointer->longest_match(string_key);
+
+  if (iter != map_pointer->end()) {
+    long counter = 0;
+    int size_of_response = iter->second.size() + iter->first.size();
+    char *return_val  = new char[size_of_response]{0};
+
+    strncpy(return_val, iter->first.c_str(), iter->first.size());
+    counter = iter->first.size();
+
```

Comment:
Instead of this `for` loop you could just do a straight `memcpy`
```suggestion
    memcpy(return_val + iter->first.size(), iter->second.data(), iter->second.size());
```

Reply:
@zindorsky that is a vector of chars .... so I don't think that works, does it?